From 7499c0e030ef5051b2920576a75dfd7c60d5232a Mon Sep 17 00:00:00 2001
From: Weijie Gao <weijie.gao@mediatek.com>
Date: Mon, 25 Jul 2022 11:18:03 +0800
Subject: [PATCH 06/30] common: board_r: add support to initialize NMBM after
 nand initialization

This patch add support to initialize NMBM after nand initialized.

Signed-off-by: Weijie Gao <weijie.gao@mediatek.com>
---
 common/board_r.c | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

--- a/common/board_r.c
+++ b/common/board_r.c
@@ -398,6 +398,20 @@ static int initr_nand(void)
 }
 #endif
 
+#ifdef CONFIG_NMBM_MTD
+
+__weak int board_nmbm_init(void)
+{
+	return 0;
+}
+
+/* go init the NMBM */
+static int initr_nmbm(void)
+{
+	return board_nmbm_init();
+}
+#endif
+
 #if defined(CONFIG_CMD_ONENAND)
 /* go init the NAND */
 static int initr_onenand(void)
@@ -713,6 +727,9 @@ static void initcall_run_r(void)
 #if CONFIG_IS_ENABLED(CMD_ONENAND)
 	INITCALL(initr_onenand);
 #endif
+#if CONFIG_IS_ENABLED(NMBM_MTD)
+	INITCALL(initr_nmbm);
+#endif
 #if CONFIG_IS_ENABLED(MMC)
 	INITCALL(initr_mmc);
 #endif
